packages/compiler/src/core/parser.ts

```diff
@@ -2474,20 +2484,23 @@ function createParser(code: string | SourceFile, options: ParseOptions = {}): Pa
         if (!newLineIsTrivia && token() === Token.NewLine) {
           break;
         }
-        if (tokenFlags() & TokenFlags.DocComment) {
-          docRanges.push({
-            pos: tokenPos(),
-            end: tokenEnd(),
-          });
-        }
+        let comment: LineComment | BlockComment | undefined = undefined;
         if (options.comments && isComment(token())) {
-          comments.push({
+          comment = {
             kind:
               token() === Token.SingleLineComment
                 ? SyntaxKind.LineComment
                 : SyntaxKind.BlockComment,
             pos: tokenPos(),
             end: tokenEnd(),
+          };
+          comments.push(comment!);
+        }
+        if (tokenFlags() & TokenFlags.DocComment) {
+          docRanges.push({
+            pos: tokenPos(),
+            end: tokenEnd(),
+            comment: comment as BlockComment,
```

Comment:
Should we make it an error to parse doc comments without saving other comments or type this as `Comment | undefined`?

Reply:
ah!